@copilot fix build errors for the .net framework build:


Error: D:\a\EPPlus\EPPlus\EPPlusTest\WorkSheetTests.cs(43,17): error CS0103: The name 'RuntimeInformation' does not exist in the current context [D:\a\EPPlus\EPPlus\EPPlusTest\EPPlusTest.Core.csproj::TargetFramework=net462]
Error: D:\a\EPPlus\EPPlus\EPPlusTest\WorkSheetTests.cs(43,49): error CS0103: The name 'OSPlatform' does not exist in the current context [D:\a\EPPlus\EPPlus\EPPlusTest\EPPlusTest.Core.csproj::TargetFramework=net462]
Error: D:\a\EPPlus\EPPlus\EPPlusTest\DrawingTest.cs(27,17): error CS0103: The name 'RuntimeInformation' does not exist in the current context [D:\a\EPPlus\EPPlus\EPPlusTest\EPPlusTest.Core.csproj::TargetFramework=net462]
Error: D:\a\EPPlus\EPPlus\EPPlusTest\DrawingTest.cs(27,49): error CS0103: The name 'OSPlatform' does not exist in the current context [D:\a\EPPlus\EPPlus\EPPlusTest\EPPlusTest.Core.csproj::TargetFramework=net462]
Error: D:\a\EPPlus\EPPlus\EPPlusTest\DTS_FailingTests.cs(19,17): error CS0103: The name 'RuntimeInformation' does not exist in the current context [D:\a\EPPlus\EPPlus\EPPlusTest\EPPlusTest.Core.csproj::TargetFramework=net462]
Error: D:\a\EPPlus\EPPlus\EPPlusTest\DTS_FailingTests.cs(19,49): error CS0103: The name 'OSPlatform' does not exist in the current context [D:\a\EPPlus\EPPlus\EPPlusTest\EPPlusTest.Core.csproj::TargetFramework=net462]
Error: D:\a\EPPlus\EPPlus\EPPlusTest\DTS_FailingTests.cs(45,17): error CS0103: The name 'RuntimeInformation' does not exist in the current context [D:\a\EPPlus\EPPlus\EPPlusTest\EPPlusTest.Core.csproj::TargetFramework=net462]
Error: D:\a\EPPlus\EPPlus\EPPlusTest\DTS_FailingTests.cs(45,49): error CS0103: The name 'OSPlatform' does not exist in the current context [D:\a\EPPlus\EPPlus\EPPlusTest\EPPlusTest.Core.csproj::TargetFramework=net462]
Error: D:\a\EPPlus\EPPlus\EPPlusTest\WorkSheetTests.cs(2283,17): error CS0103: The name 'RuntimeInformation' does not exist in the current context [D:\a\EPPlus\EPPlus\EPPlusTest\EPPlusTest.Core.csproj::TargetFramework=net462]
Error: D:\a\EPPlus\EPPlus\EPPlusTest\WorkSheetTests.cs(2283,49): error CS0103: The name 'OSPlatform' does not exist in the current context [D:\a\EPPlus\EPPlus\EPPlusTest\EPPlusTest.Core.csproj::TargetFramework=net462]
Error: D:\a\EPPlus\EPPlus\EPPlusTest\WorkSheetTests.cs(2298,17): error CS0103: The name 'RuntimeInformation' does not exist in the current context [D:\a\EPPlus\EPPlus\EPPlusTest\EPPlusTest.Core.csproj::TargetFramework=net462]
Error: D:\a\EPPlus\EPPlus\EPPlusTest\WorkSheetTests.cs(2298,49): error CS0103: The name 'OSPlatform' does not exist in the current context [D:\a\EPPlus\EPPlus\EPPlusTest\EPPlusTest.Core.csproj::TargetFramework=net462]

(check the csproj/workflow to see which platforms it's supposed to be running on)

--- a/EPPlusTest/DTS_FailingTests.cs
+++ b/EPPlusTest/DTS_FailingTests.cs
@@ -15,7 +15,7 @@ namespace EPPlusTest
         [TestMethod]
         public void DeleteWorksheetWithReferencedImage()
         {
-            if (Environment.OSVersion.Platform == PlatformID.Unix)
+            if (Environment.OSVersion.Platform == PlatformID.Unix && Directory.Exists("/proc"))
             {
                 Assert.Inconclusive("Skipping image worksheet tests on Linux.");
             }
@@ -41,7 +41,7 @@ namespace EPPlusTest
         [TestMethod]
         public void CopyAndDeleteWorksheetWithImage()
         {
-            if (Environment.OSVersion.Platform == PlatformID.Unix)
+            if (Environment.OSVersion.Platform == PlatformID.Unix && Directory.Exists("/proc"))
             {
                 Assert.Inconclusive("Skipping image worksheet tests on Linux.");
             }

--- a/EPPlusTest/DTS_FailingTests.cs
+++ b/EPPlusTest/DTS_FailingTests.cs
@@ -15,7 +15,7 @@ namespace EPPlusTest
         [TestMethod]
         public void DeleteWorksheetWithReferencedImage()
         {
-            if (Environment.OSVersion.Platform == PlatformID.Unix && Directory.Exists("/proc"))
+            if (TestPlatform.IsLinux)
             {
                 Assert.Inconclusive("Skipping image worksheet tests on Linux.");
             }
@@ -41,7 +41,7 @@ namespace EPPlusTest
         [TestMethod]
         public void CopyAndDeleteWorksheetWithImage()
         {
-            if (Environment.OSVersion.Platform == PlatformID.Unix && Directory.Exists("/proc"))
+            if (TestPlatform.IsLinux)
             {
                 Assert.Inconclusive("Skipping image worksheet tests on Linux.");
             }

--- a/EPPlusTest/DTS_FailingTests.cs
+++ b/EPPlusTest/DTS_FailingTests.cs
@@ -5,7 +5,6 @@ using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using System.Text;
-using System.Runtime.InteropServices;
 
 namespace EPPlusTest
 {
@@ -16,7 +15,7 @@ namespace EPPlusTest
         [TestMethod]
         public void DeleteWorksheetWithReferencedImage()
         {
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            if (Environment.OSVersion.Platform == PlatformID.Unix)
             {
                 Assert.Inconclusive("Skipping image worksheet tests on Linux.");
             }
@@ -42,7 +41,7 @@ namespace EPPlusTest
         [TestMethod]
         public void CopyAndDeleteWorksheetWithImage()
         {
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            if (Environment.OSVersion.Platform == PlatformID.Unix)
             {
                 Assert.Inconclusive("Skipping image worksheet tests on Linux.");
             }

--- a/EPPlusTest/DrawingTest.cs
+++ b/EPPlusTest/DrawingTest.cs
@@ -23,7 +23,7 @@ namespace EPPlusTest
         [TestMethod]
         public void RunDrawingTests()
         {
-            if (Environment.OSVersion.Platform == PlatformID.Unix && Directory.Exists("/proc"))
+            if (TestPlatform.IsLinux)
             {
                 Assert.Inconclusive("Skipping drawing tests on Linux.");
             }

--- a/EPPlusTest/DrawingTest.cs
+++ b/EPPlusTest/DrawingTest.cs
@@ -23,7 +23,7 @@ namespace EPPlusTest
         [TestMethod]
         public void RunDrawingTests()
         {
-            if (Environment.OSVersion.Platform == PlatformID.Unix)
+            if (Environment.OSVersion.Platform == PlatformID.Unix && Directory.Exists("/proc"))
             {
                 Assert.Inconclusive("Skipping drawing tests on Linux.");
             }

--- a/EPPlusTest/DrawingTest.cs
+++ b/EPPlusTest/DrawingTest.cs
@@ -10,7 +10,6 @@ using OfficeOpenXml.Drawing.Chart;
 using OfficeOpenXml.Style;
 using System.Diagnostics;
 using System.Reflection;
-using System.Runtime.InteropServices;
 
 namespace EPPlusTest
 {
@@ -24,7 +23,7 @@ namespace EPPlusTest
         [TestMethod]
         public void RunDrawingTests()
         {
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            if (Environment.OSVersion.Platform == PlatformID.Unix)
             {
                 Assert.Inconclusive("Skipping drawing tests on Linux.");
             }

--- a/EPPlusTest/TestBase.cs
+++ b/EPPlusTest/TestBase.cs
@@ -6,6 +6,11 @@ using System.Reflection;
 
 namespace EPPlusTest
 {
+    internal static class TestPlatform
+    {
+        internal static bool IsLinux => Environment.OSVersion.Platform == PlatformID.Unix && Directory.Exists("/proc");
+    }
+
     [TestClass]
     public abstract class TestBase
     {

--- a/EPPlusTest/WorkSheetTests.cs
+++ b/EPPlusTest/WorkSheetTests.cs
@@ -39,7 +39,7 @@ namespace EPPlusTest
         [TestMethod]
         public void RunWorksheetTests()
         {
-            if (Environment.OSVersion.Platform == PlatformID.Unix)
+            if (Environment.OSVersion.Platform == PlatformID.Unix && Directory.Exists("/proc"))
             {
                 Assert.Inconclusive("Skipping worksheet integration tests on Linux.");
             }
@@ -2279,7 +2279,7 @@ namespace EPPlusTest
         [TestMethod]
         public void SetBackground()
         {
-            if (Environment.OSVersion.Platform == PlatformID.Unix)
+            if (Environment.OSVersion.Platform == PlatformID.Unix && Directory.Exists("/proc"))
             {
                 Assert.Inconclusive("Skipping drawing-related worksheet tests on Linux.");
             }
@@ -2294,7 +2294,7 @@ namespace EPPlusTest
         [TestMethod]
         public void SetHeaderFooterImage()
         {
-            if (Environment.OSVersion.Platform == PlatformID.Unix)
+            if (Environment.OSVersion.Platform == PlatformID.Unix && Directory.Exists("/proc"))
             {
                 Assert.Inconclusive("Skipping drawing-related worksheet tests on Linux.");
             }

--- a/EPPlusTest/WorkSheetTests.cs
+++ b/EPPlusTest/WorkSheetTests.cs
@@ -16,7 +16,6 @@ using OfficeOpenXml.Table;
 using System.Threading;
 using System.Globalization;
 using System.Threading.Tasks;
-using System.Runtime.InteropServices;
 
 namespace EPPlusTest
 {
@@ -40,7 +39,7 @@ namespace EPPlusTest
         [TestMethod]
         public void RunWorksheetTests()
         {
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            if (Environment.OSVersion.Platform == PlatformID.Unix)
             {
                 Assert.Inconclusive("Skipping worksheet integration tests on Linux.");
             }
@@ -2280,7 +2279,7 @@ namespace EPPlusTest
         [TestMethod]
         public void SetBackground()
         {
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            if (Environment.OSVersion.Platform == PlatformID.Unix)
             {
                 Assert.Inconclusive("Skipping drawing-related worksheet tests on Linux.");
             }
@@ -2295,7 +2294,7 @@ namespace EPPlusTest
         [TestMethod]
         public void SetHeaderFooterImage()
         {
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            if (Environment.OSVersion.Platform == PlatformID.Unix)
             {
                 Assert.Inconclusive("Skipping drawing-related worksheet tests on Linux.");
             }

--- a/EPPlusTest/WorkSheetTests.cs
+++ b/EPPlusTest/WorkSheetTests.cs
@@ -39,7 +39,7 @@ namespace EPPlusTest
         [TestMethod]
         public void RunWorksheetTests()
         {
-            if (Environment.OSVersion.Platform == PlatformID.Unix && Directory.Exists("/proc"))
+            if (TestPlatform.IsLinux)
             {
                 Assert.Inconclusive("Skipping worksheet integration tests on Linux.");
             }
@@ -2279,7 +2279,7 @@ namespace EPPlusTest
         [TestMethod]
         public void SetBackground()
         {
-            if (Environment.OSVersion.Platform == PlatformID.Unix && Directory.Exists("/proc"))
+            if (TestPlatform.IsLinux)
             {
                 Assert.Inconclusive("Skipping drawing-related worksheet tests on Linux.");
             }
@@ -2294,7 +2294,7 @@ namespace EPPlusTest
         [TestMethod]
         public void SetHeaderFooterImage()
         {
-            if (Environment.OSVersion.Platform == PlatformID.Unix && Directory.Exists("/proc"))
+            if (TestPlatform.IsLinux)
             {
                 Assert.Inconclusive("Skipping drawing-related worksheet tests on Linux.");
             }

